src/modules/sensors/vehicle_gps_position/VehicleGPSPosition.cpp

```diff
@@ -95,54 +100,62 @@ VehicleGPSPosition::Run()
 {
 	ParametersUpdate();
 
-	// read gps1 data if available
-	bool gps1_updated = _sensor_gps_sub[0].updated();
+	// Check all GPS instance
+	bool any_gps_updated = false;
+	bool gps_updated[GPS_MAX_RECEIVERS];
 
-	if (gps1_updated) {
-		_sensor_gps_sub[0].copy(&_gps_state[0]);
-	}
+	for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
 
-	// check for second GPS receiver data
-	bool gps2_updated = _sensor_gps_sub[1].updated();
+		gps_updated[i] = _sensor_gps_sub[i].updated();
 
-	if (gps2_updated) {
-		_sensor_gps_sub[1].copy(&_gps_state[1]);
+		if (gps_updated[i]) {
+			any_gps_updated = true;
+			_sensor_gps_sub[i].copy(&_gps_state[i]);
+		}
 	}
 
-	if ((_param_sens_gps_mask.get() == 0) && gps1_updated) {
+
+	if ((_param_sens_gps_mask.get() == 0) && gps_updated[0]) {
 		// When GPS blending is disabled we always use the first receiver instance
 		Publish(_gps_state[0]);
 
-	} else if ((_param_sens_gps_mask.get() > 0) && (gps1_updated || gps2_updated)) {
-		// blend dual receivers if available
+	} else if ((_param_sens_gps_mask.get() > 0) && any_gps_updated) {
+		// blend multiple receivers if available
 
 		// calculate blending weights
 		if (!blend_gps_data()) {
-			// handle case where the blended states cannot be updated
-			if (_gps_state[0].fix_type > _gps_state[1].fix_type) {
-				// GPS 1 has the best fix status so use that
-				_gps_select_index = 0;
-
-			} else if (_gps_state[1].fix_type > _gps_state[0].fix_type) {
-				// GPS 2 has the best fix status so use that
-				_gps_select_index = 1;
-
-			} else if (_gps_select_index == 2) {
-				// use last receiver we received data from
-				if (gps1_updated) {
-					_gps_select_index = 0;
-
-				} else if (gps2_updated) {
-					_gps_select_index = 1;
+			// Only use selected receiver data if it has been updated
+			_gps_new_output_data = false;
+			_gps_select_index = 0;
+
+			// Find the single "best" GPS from the data we have
+			// First, find the GPS(s) with the best fix
+			uint8_t best_fix = 0;
+
+			for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+				if (_gps_state[i].fix_type > best_fix) {
+					best_fix = _gps_state[i].fix_type;
 				}
 			}
 
-			// Only use selected receiver data if it has been updated
-			if ((gps1_updated && _gps_select_index == 0) || (gps2_updated && _gps_select_index == 1)) {
-				_gps_new_output_data = true;
+			// Second, compare GPS's with best fix and take the one with most satellites
+			uint8_t max_sats = 0;
```

Comment:
This was just the quickest/simplest heuristic I came up with for picking which GPS was best; we could also use an accuracy measurement if all GPSs are publishing compatible quantities.

Reply:
Is it possible for a GPS unit to report "No fix" (fix_type == 0) while still reporting some number of satellites?

Reply:
And vice versa, reporting "3D fix" while having zero satellites?

Reply:
If one GPS doesn't have a fix while another does, this will automatically discard the GPS without the fix (only the GPSs with the "best fix" will be sorted through).

Reply:
And as to 0 satellites... I doubt that would happen, but I could see a case where two different units both have a 3D fix with multiple satellites, but the one with fewer satellites could have a better speed/position accuracy estimate than the one with more satellites.  Unlikely perhaps, but plausible.

Reply:
@dakejahl could you add me as a co-author on this commit, seeing as I did the original blending modifications here?

Reply:
I've likely already made a mess of things with squashed commits. Happy to restore credit.